test/support/publish_dsl.rb

```diff
@@ -25,11 +25,191 @@ def publish_wait
+        visit series_url
+        click_button('Delete')
+        click_button('Okay')
+      end
+      visit series_list_url
+    end
+
+    def delete_all_episodes_in_series!(series_url)
+      episode_list_url = series_url + '/list'
+      visit episode_list_url
+      begin
+        find('a', text: /Search among these episodes/)
+      rescue
+        return # no episodes
+      end
+      episode_links = {}
+      page.all('a', text: /^Test Episode \w{20}$/).each do |link|
+        episode_url = link[:href]
+        next if episode_links[episode_url]
+        episode_links[episode_url] = true
+      end
+      episode_links.keys.each do |episode_url|
+        puts "Deleting #{episode_url}" if debug?
+        m = episode_url.match(/\/story\/(\d+)$/)
+        ep_id = m[1]
+        # there's a wonky bug with Poltergeist and Angular where the modal Okay
+        # button does not trigger a DELETE call. So we do it ourselves.
+        puts "Episode ID: #{ep_id}" if debug?
+        del_url = 'UNKNOWN'
+        authz_token = 'UNKNOWN'
+        page.driver.network_traffic.each do |req|
+          if req.url.match(/authorization$/)
+            del_url = req.url + "/stories/#{ep_id}"
+            req.headers.each do |h|
+              if h['name'] == 'Authorization'
+                authz_token = h['value']
+                break
+              end
+            end
+          end
+          break if authz_token != 'UNKNOWN'
+        end
+        puts "DELETE #{del_url}" if debug?
+        page.execute_script(publish_js_to_delete_episode(del_url, authz_token))
+      end
+      sleep(1) # time to finish XHR
+    end
+
+    def publish_js_to_delete_episode(del_url, authz_token)
+      debugger_line = debug? ? "console.log('DELETE ok')" : ''
+      <<~JSCRIPT
+        var xhr=new XMLHttpRequest();
+        xhr.open('DELETE','#{del_url}');
+        xhr.setRequestHeader('Authorization', '#{authz_token}');
+        xhr.setRequestHeader('Accept', 'application/hal+json');
+        xhr.onload=function() {
+          if (xhr.status === 204) {
+            #{debugger_line}
+          } else {
+            console.log('XHR fail:' + xhr.status)
+          }
+        };
+        xhr.withCredentials = true;
+        xhr.send();
+      JSCRIPT
+    end
+
+    def create_series!(random_str = SecureRandom.hex(10))
+      publish_login!
+      visit CONFIG.PUBLISH_HOST
+      click_link 'New Series'
+      fill_in 'Series Title', with: "Test Series #{random_str}"
+      fill_in 'Teaser', with: "Test Series Teaser #{random_str}"
+      click_button 'Create'
+      publish_wait_for_success
+      publish_wait_for_title
+      page.must_have_content("Test Series #{random_str}")
+      page.must_have_content('Podcast Info')
+
+      publish_wait_for_episodes_link
+      page.must_have_content("0 Episodes in Series")
+
+      click_link 'Podcast Info'
+      click_button 'Create Podcast'
+      # TODO how to select dropdown checkbox?
```

Comment:
I've found myself working around this question a bit too. I'll let you know if I find out how to select that dropdown via DOM.

Reply:
We could probably do a better job with our interactive modals/forms/etc.  Might be worth spending some time pointing these tests at publish.prx.dev, and working through some of them.